src/Room.js

```diff
@@ -138,6 +139,21 @@ class Room extends GameEntity {
     item.room = null;
   }
 
+  /**
+   * Check if diagonal directions are enabled
+   *
+   * @return {boolean}
+   */
+  checkDiagonalDirections() {
```

Comment:
This name could be a bit clearer, something like `checkDiagonalDirectionsEnabled` or `hasDiagonalDirectionsEnabled` -- which tells us that it is checking a flag, not the directions themselves, and that it will return true if they are enabled.